Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/EdgeSetGeometryAlgorithms.inl

```diff
@@ -361,44 +361,28 @@ bool EdgeSetGeometryAlgorithms<DataTypes>::isPointOnEdge(const sofa::type::Vec<3
     return sofa::geometry::Edge::isPointOnEdge(pt, p1, p2);
 }
 
-//
+
 template<class DataTypes>
-auto EdgeSetGeometryAlgorithms<DataTypes>::compute2PointsBarycoefs(
+auto EdgeSetGeometryAlgorithms<DataTypes>::computeEdgeBarycentricCoordinates(
     const sofa::type::Vec<3, Real> &p,
-    PointID ind_p1,
-    PointID ind_p2) const -> sofa::type::vector< SReal >
+    PointID ind_p1, PointID ind_p2, bool useRestPosition) const -> sofa::type::vector< SReal >
 {
-    constexpr Real ZERO = static_cast<Real>(1e-6);
-
-    sofa::type::vector< SReal > baryCoefs;
+    sofa::core::ConstVecCoordId::MyVecId _vecId = useRestPosition ? core::ConstVecCoordId::restPosition() : core::ConstVecCoordId::position();
 
-    const typename DataTypes::VecCoord& vect_c =(this->object->read(core::ConstVecCoordId::position())->getValue());
+    const typename DataTypes::VecCoord& vect_c = (this->object->read(_vecId)->getValue());
```

Comment:
minor suggestion but why a prefix (with _) as it is is just a variable ?

Reply:
was just to differentiate from any possible type: VecId  but I am open to any suggestion.